libraries/consul_acl.rb

```diff
@@ -0,0 +1,65 @@
+# Cookbook: consul
+# License: Apache 2.0
+#
+# Copyright (C) 2014, 2015 Bloomberg Finance L.P.
+#
+require 'poise'
+
+module ConsulCookbook
+  module Resource
+    # @since 1.0.0
+    class ConsulACL < Chef::Resource
+      include Poise(fused: true)
+      provides(:consul_acl)
+      default_action(:create)
+
+      # @!attribute name
+      # @return [String]
+      attribute(:name, name_attribute: true, kind_of: String)
+      # @!attribute rules
+      # @return [String]
+      attribute(:rules, kind_of: String)
+      # @!attribute id
+      # @return [String]
+      attribute(:id, kind_of: String)
+      # @!attribute type
+      # @return [String]
+      attribute(:type, kind_of: String, default: 'client', required: true)
+
+      action :create do
+        ruby_block @new_resource.name do
+          data = {}
+          data['Name'] = new_resource.name
+          data['Type'] = new_resource.type
+          data['Rules'] = new_resource.rules
+          data['ID'] = new_resource.id unless new_resource.id.nil?
+          block { client.manage_acl(data) }
+          only_if { !current_resource || update_required? }
+        end
+      end
+
+      action :delete do
+        ruby_block new_resource.name do
+          block { client.delete_acl(new_resource.name) }
+          only_if { current_resource }
+        end
+      end
+
+      def client
+        @client ||= Chef::Consul::Client.new(nil, node['consul']['ports']['http'], node['consul']['acl_master_token'])
```

Comment:
Please pass in the node attributes as properties (attributes) on the resource. This allows us to validate at convergence time and also makes definition of resource more explicit.